### PR TITLE
don't fail check-param-names on destructuring in arguments - fixes #90

### LINKED ISSUE #90: destructuring parameters fail checkParamNames
These days I'm often using the following pattern to pass variables to functions
```js
var f = function ({prop1, prop2}) {
  return prop1 + prop2;
};
```

This leads to the problem, that the destructuring object doesn't have a name, which is required by ```checkParamNames```.

Giving the parameter no name, leads to the ```Missing param name``` error
```js
/**
 * [f description]
 *
 * @param  {Object}
---^
 * @return {String} Some string
 */
```

Giving it a random name leads to the ```Expected undefined but got object``` error.
```js
/**
 * [f description]
 *
 * @param  {Object} object
--------------------^
 * @return {String} Some string
 */
```

Trying to hack it with undefined gives a ```Parameters undefined and undefined are out of order``` error.
```js
/**
 * [f description]
 *
 * @param  {Object} undefined
--------------------^
 * @return {String} Some string
 */
```

My .jscsrc:

```json
  "jsDoc": {
    "checkParamNames": true,
    "checkRedundantParams": true,
    "requireParamTypes": true,
    "checkReturnTypes": true,
    "checkRedundantReturns": true,
    "requireReturnTypes": true,
    "checkTypes": true,
    "checkRedundantAccess": true,
    "leadingUnderscoreAccess": "private",
    "enforceExistence": true,
    "checkAnnotations": true,
    "requireReturnTypes": true,
    "checkRedundantAccess": true,
    "requireHyphenBeforeDescription": true
  },
```

Is there a way I can get this to work properly? If there is nothing implemented right now, I would suggest automatically disabling the ```checkParamNames``` rule for destructuring parameters. Thoughts?

### PR DESCRIPTION
**Pull request by alawatthe**
Here is a first stab at fixing #90.

I had to create a new ```describe``` block in the tests, since the code which should be tested requires the ```esnext``` flag for jscs.

Let me know what you think :-)

### CODE REVIEW

File: test/lib/rules/validate-jsdoc/check-param-names.js [hunk 1]
```diff
@@ -367,4 +367,61 @@ describe('lib/rules/validate-jsdoc/check-param-names', function() {
 
     });
 
+
```
Comment (zxqfox) [inline test/lib/rules/validate-jsdoc/check-param-names.js]: Can you drop one empty line?

File: test/lib/rules/validate-jsdoc/check-param-names.js [hunk 2]
```diff
@@ -367,4 +367,61 @@ describe('lib/rules/validate-jsdoc/check-param-names', function() {
...
+                    'function arr([param]) {',
+                    '}'
+                ].join('\n')
+            }
+            // jscs:enable
+            /* jshint ignore:end */
+        ]);
+
+    });
 });
```
Comment (zxqfox) [inline test/lib/rules/validate-jsdoc/check-param-names.js]: And add one here ;-)

**Comment by zxqfox** [comment]
:+1: I like it ;-) Thanks for contribution!

I don't like this part of code at all. But this PR is pretty fine.

**Comment by alawatthe** [comment]
Sorry for the delay, I was really busy :-/

I inserted/removed the lines. Furthermore I added support for fake parameter names, i.e. this example is now checked properly:

```js
/**
 * My cool function.
 *
 * @param {Object} fakeName - An object.
 * @param {string} fakeName.prop1 - Property 1.
 * @param {string} fakeName.prop2 - Property 2.
 */
var fn = function ({prop1, prop2}) {
  // Do something with prop1 and prop2
}
```

**Comment by zxqfox** [comment]
Oh, wow. Looks great ;-)

I'm gonna merge this soon after some works.

**Pull request closed without merging**